src/main/java/com/mraof/minestuck/blockentity/machine/PunchDesignixBlockEntity.java

```diff
@@ -152,16 +155,16 @@ private void handleKeyboardClick(Player player)
 			}
 		} else
 		{
-			if(AlchemyHelper.isReadableCard(heldStack))
+			if(heldStack.is(MSItems.CAPTCHA_CARD) && !heldStack.has(MSItemComponents.ENCODED_ITEM))
 			{
-				setCaptcha(CardCaptchas.getCaptcha(AlchemyHelper.getDecodedItem(heldStack).getItem(), player.getServer()));
-				effects(false);
-			} else if(heldStack.is(MSItems.CAPTCHA_CARD.get()) && AlchemyHelper.getDecodedItem(heldStack).isEmpty())
-			{
-				setCaptcha(CardCaptchas.EMPTY_CARD_CAPTCHA);
-				effects(false);
-			} else if(heldStack.is(MSItems.CAPTCHA_CARD.get()) && !AlchemyHelper.isReadableCard(heldStack))
-				player.displayClientMessage(Component.translatable(REJECT_CARD), true); //card unreadable
+				CardStoredItemComponent cardStoredItemComponent = heldStack.getOrDefault(MSItemComponents.CARD_STORED_ITEM, CardStoredItemComponent.EMPTY);
+				if(cardStoredItemComponent.code() != null)
+				{
+					setCaptcha(cardStoredItemComponent.code());
+					effects(false);
+				} else
+					player.displayClientMessage(Component.translatable(REJECT_CARD), true); //card unreadable
+			}
```

Comment:
In prior game versions, using punched cards successfully sets the captcha. Realistically players shouldnt have been able to "read" the punched card to begin with but it is still a quality of life feature we may want to discuss returning

Reply:
I'm not seeing this feature when looking in the code for 1.20.1, so I'm not sure when this last was a feature.
It should be a fairly easy feature to implement here. But to avoid needing to determine  when a discussion has been had and a decision has been made, I'm inclined to keep it in line with the behavior in the last update(s) for this PR.